packages/sitecore-jss-nextjs/src/components/NextImage.tsx

```diff
@@ -13,31 +13,33 @@ import Image, {
   ImageProps as NextImageProperties,
 } from 'next/image';
 
-type NextImageProps = Pick<
-  ImageProps,
-  'editable' | 'imageParams' | 'field' | 'mediaUrlPrefix' | 'srcSet'
-> &
-  Partial<NextImageProperties>;
+type NextImageProps = Omit<ImageProps, 'media'> & Partial<NextImageProperties>;
 
 export type ImageConfigComplete = {
   path: string;
 };
 
-export type ImageConfig = Partial<ImageConfigComplete>;
-
-const { path: configPath } = (process.env.__NEXT_IMAGE_OPTS as any) as ImageConfigComplete;
-
 export const loader: ImageLoader = ({ src, width }: ImageLoaderProps): string => {
-  try {
-    const url = new URL(`${configPath}${src}`);
-    const params = url.searchParams;
-    params.set('mw', params.get('mw') || width.toString());
-    params.delete('w');
+  const { path: configPath } = (process.env.__NEXT_IMAGE_OPTS as any) as ImageConfigComplete;
 
-    return url.href;
+  try {
+    const r = /^(?:[a-z]+:)?\/\//i;
+    if (r.test(src)) {
+      const url = new URL(`${src}`);
+      const params = url.searchParams;
+      params.set('mw', params.get('mw') || width.toString());
+      params.delete('w');
+      return url.href;
+    } else {
+      const url = new URL(`${configPath}${src}`);
+      const params = url.searchParams;
+      params.set('mw', params.get('mw') || width.toString());
+      params.delete('w');
+      return url.href;
+    }
```

Comment:
Can reduce code duplication here. i.e.
```suggestion
    const r = /^(?:[a-z]+:)?\/\//i;
    const url = r.test(src) ? new URL(`${src}`) : new URL(`${configPath}${src}`);
    const params = url.searchParams;
    params.set('mw', params.get('mw') || width.toString());
    params.delete('w');
    return url.href;
```